packages/notebook/src/widget.ts

```diff
@@ -1028,46 +1037,26 @@ export class StaticNotebook extends WindowedList<NotebookViewModel> {
 
     // Apply content visibility when notebook widget is attached to the DOM
     if (this._notebookConfig.windowingMode === 'contentVisibility') {
-      // Apply content visibility to all cells initially
-      this._applyContentVisibilityToAllCells();
+      this.toggleClass('jp-content-visibility', true);
 
-      // watch for newly added cells
-      this.model?.cells.changed.connect(() => {
-        this._applyContentVisibilityToAllCells();
+      // Update intrinsic sizes for all cells initially
+      requestAnimationFrame(() => {
+        this.cellsArray.forEach((cell, i) => {
+          const estHeight = this._viewModel.estimateWidgetSize(i);
+          cell.node.style.containIntrinsicSize = `auto ${estHeight}px`;
+        });
       });
```

Comment:
This looks like a duplication with `_runOnIdleTime` way of doing it?

Reply:
Yes, but would not running that on idle time eliminate the need for this iteration here?